### PR TITLE
Add storage mode flag

### PR DESCRIPTION
**Pull request by as51340**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: src/flags/storage_mode.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+#include <array>
+
+inline constexpr std::array storage_mode_mappings{
+    std::pair{std::string_view{"IN_MEMORY_TRANSACTIONAL"}, memgraph::storage::StorageMode::IN_MEMORY_TRANSACTIONAL},
+    std::pair{std::string_view{"IN_MEMORY_ANALYTICAL"}, memgraph::storage::StorageMode::IN_MEMORY_ANALYTICAL},
+    std::pair{std::string_view{"ON_DISK_TRANSACTIONAL"}, memgraph::storage::StorageMode::ON_DISK_TRANSACTIONAL}};
+
+const std::string storage_mode_help_string =
+    fmt::format("Storage mode Memgraph uses. Allowed values: {}",
```
Comment (Ignition) [inline src/flags/storage_mode.cpp:28]: More correct to say the "default" storage mode memgraph uses.

File: src/dbms/database.cpp [hunk 2]
```diff
@@ -20,10 +22,15 @@ namespace memgraph::dbms {
 Database::Database(const storage::Config &config)
     : trigger_store_(config.durability.storage_directory / "triggers"),
       streams_{config.durability.storage_directory / "streams"} {
-  if (config.force_on_disk || utils::DirExists(config.disk.main_storage_directory)) {
+  auto storage_mode = memgraph::flags::ParseStorageMode();
```
Comment (Ignition) [inline src/dbms/database.cpp:25]: Is it worth while putting it into the storage::Config as `default_storage_mode`

File: src/dbms/database.cpp [hunk 3]
```diff
@@ -20,10 +22,15 @@ namespace memgraph::dbms {
...
+  if (storage_mode == memgraph::storage::StorageMode::ON_DISK_TRANSACTIONAL || config.force_on_disk ||
+      utils::DirExists(config.disk.main_storage_directory)) {
     storage_ = std::make_unique<storage::DiskStorage>(config);
   } else {
     storage_ = std::make_unique<storage::InMemoryStorage>(config);
+    if (storage_mode == memgraph::storage::StorageMode::IN_MEMORY_ANALYTICAL) {
+      storage_->SetStorageMode(storage_mode);
+    }
```
Comment (Ignition) [inline src/dbms/database.cpp:33]: What about changing the InMemoryStorage constructor to use `default_storage_mode` from config?

File: src/dbms/database.cpp (hunk 2 above)
Comment (as51340) [reply src/dbms/database.cpp:25]: You mean why parse, when I can put the flag into storage::Config or?

File: src/dbms/database.cpp (hunk 3 above)
Comment (as51340) [reply src/dbms/database.cpp:33]: hmm I don't understand: why would you change InMemoryStorage constructor

Comment (Ignition) [reply src/dbms/database.cpp:33]: `storage::InMemoryStorage` is constructed then, immediately the default storage mode that constructor uses wasn't good enough, hence you call `SetStorageMode`. Why not make it the constructor responsibility.

**Review by Ignition — APPROVED** [review]

**Comment by as51340** [comment]
Docs: https://github.com/memgraph/documentation/pull/220
Release note: You can start Memgraph with any storage mode using --storage-mode.

**Pull request merged**